### ISSUE TITLE
elements onClick not triggered in draggable-state on phone

### ISSUE THREAD

**Issue by peteranderss0n** [react-grid-layout/react-grid-layout#1793]
### Describe the bug

When you are on a touch device, react elements onClick won't trigger when you click on them when `isDraggable={true}` are enabled. It works as intended in desktop.

In previous versions this worked on touch-devices, here are two CodeSandboxes illustrating the difference;

_Not working on touch:_
https://codesandbox.io/s/sweet-cerf-kkh0wf?file=/src/ShowcaseLayout.js

_Old version that works on touch:_
https://codesandbox.io/s/wandering-voice-oqf511?file=/src/ShowcaseLayout.js

From what I can see, this support was removed in version 0.17.

### Your Example Website or App

https://codesandbox.io/s/sweet-cerf-kkh0wf?file=/src/ShowcaseLayout.js

### Steps to Reproduce the Bug or Issue

1. Go to sandbox on a touch device or enable "toggle device toolbar" simulation in Chrome
2. Click on the h1 - "test"

### Expected behavior

Expected to be alerted with "test" since it has a onClick attached to it.

### react-grid-layout library version

1.3.4

### Operating System Version

Windows

### Browser

Chrome

**Comment by lorenzobalzani** [comment]
I am experiencing the same issue. Were you able to solve this problem?

**Comment by peteranderss0n** [comment]
@lorenzobalzani the way I worked around it was to attach two click events. One normal onClick() and one 
`onTouchEnd={() => { //Do something }}`

**Comment by lorenzobalzani** [comment]
> @lorenzobalzani the way I worked around it was to attach two click events. One normal onClick() and one for touch-screens 
> 
> ```
> 
> onTouchEnd={() => { 
> 
>     //Do something 
> 
> }}
> 
> ```

Thank you 😉

**Comment by stephenrs** [comment]
A quick follow-up. I'm using v1.3.4, so I'm wondering if this issue has been addressed in later versions.

I have a fairly complex UI, and having to implement onTouchStart explicitly for every grid item that needs to receive onClick events is a bit messy and painful. It seems that react-grid-layout (v1.3.4) is fighting against built-in React functionality.

**Comment by stephenrs** [comment]
Can @STRML or someone else connected to the project please comment on this? I feel like I must be missing something. Is it the intention that items that are part of a react-grid-layout should not be interactive on mobile/touch devices?

The OP was correct that items stopped receiving click/touch events with the release of 0.17 and it appears to have been a permanent change because the latest version still exhibits the same behavior.

This continues to be a problem for my project because some of my grid items are rendered using external components/libraries, so the only way for me to make them work with recent versions of RGL would be to fork/hack the original code to listen for touch events (or fork/hack RGL, I guess).

I'd greatly appreciate any information that you can provide.

**Comment by stephenrs** [comment]
From what I can tell, this commit introduced the problem: https://github.com/react-grid-layout/react-draggable/commit/8a51de911fccb282d8444641dbd1e7b6d92ae603

That line of code might fix the scrolling problem reported on discussed here: https://github.com/react-grid-layout/react-draggable/pull/183, but it breaks all grid items that expect to receive touch events on mobile, rendering them useless. Commenting out that line fixes the problem reported on this issue.

So, since the scrolling problem doesn't affect my project, it looks like I now have a private forked version of RGL that comments out the line that was added in the above commit. I would obviously love to see the official version of RGL work fully on mobile again, however. 

Hopefully this helps someone.

**Comment by STRML** [comment]
> *This was generated by AI during triage.*

- Touch onClick suppressed

## Agent Brief

**Category:** bug
**Summary:** On touch devices, grid-item children lose their `onClick` whenever `isDraggable` is true, because `DraggableCore` calls `e.preventDefault()` on `touchstart` (react-draggable's `handleDragStart`, gated by `allowMobileScroll`, default false), and v2 passes no `allowMobileScroll` through.

**Current behavior:**
`GridItem` renders `<DraggableCore disabled={!isDraggable} ...>` with no `allowMobileScroll`. The `DragConfig` type and defaults don't accept the flag. The native `click` after a preventDefault-ed `touchstart` is suppressed, so children's `onClick` never fires on touch.

**Desired behavior:**
Expose `allowMobileScroll` on `dragConfig` (v2) and as a legacy flat prop, thread it into `DraggableCore`. Decide the default: keep `false` for scroll-prevention parity and document the trade-off, or default `true` and document that consumers who need touch scroll can. Document the choice in the README/API docs.

**Key interfaces:**
- `DragConfig` (v2) - add `allowMobileScroll?: boolean`
- `GridItem` - pass it to `<DraggableCore>`
- Legacy wrapper (`ReactGridLayout` / `ResponsiveReactGridLayout`) - forward the flat prop

**Acceptance criteria:**
- [ ] A simulated touchstart/touchmove/touchend/click shows a child `onClick` fires when the flag is on
- [ ] The legacy flat prop forwards correctly
- [ ] `yarn lint`, `yarn fmt`, `make test` green

**Out of scope:**
- Changing react-draggable's default
- Making non-draggable items suppress scroll (they already don't)

**Comment by STRML** [comment]
> *This was generated by AI during triage.*

Status update on the brief posted 2026-08-04: the bulk of this landed in f13dbbc — `DragConfig.allowMobileScroll`, `GridItem` → `DraggableCore` threading, the legacy `ReactGridLayout` flat prop, and forwarding tests in `test/spec/touch-drag-props-test.tsx`.

One gap remained against the brief's acceptance criteria: `ResponsiveReactGridLayout` rebuilt `dragConfig` without the flat prop, so it silently died on the responsive wrapper. #2289 threads it through and extends the test coverage. Keeping this open until #2289 merges, then closing as fixed (ships after 2.2.4).
